documentation/developers.md

```diff
@@ -65,6 +65,42 @@ might be implemented to utilize the `compare-levels` function:
                                        :z "y"))}))
 ```
 
+### Implementing Alerts
+
+Alerts are returned by the  status function and allow you to expose human
+readable information about the state of your application. This makes it easy
+for status to be displayed and checked by unopinionated consumers. This works
+best if all conditions which may effect the availability of your service
+result in an alert.
+
+Examples:
+```clj
+{:severity :error
+ :message "Cannot connect to database"
+ :type :my-service/databas-unavailable
+ :details {}}
+
+{:severity :warning
+ :message "Repo may not be initialized"
+ :type :my-service/uninitilized-repo
+ :details { :repo "some-repo"} }
+```
+
+- `:severity` : The severity of the alert represents how consumers should support it.
+    - `:error` Should be used if an only if the alert is serious to prevent the
+      service from being in the `:running` state.
+    - `:warning` Should be used for conditions which will likely result in
+      degraded service without effecting the `:state`
+    - `:info` : Should be used for other alerts that do not impact the
+      functioning of the service. These should not be present when the status
+      level is greater than info.
+- `:message` : A human readable string that can be displayed to users.
+- `:type` : This can be used by consumer logic to find particular alerts that
+  are important to them.
+- `:details` : Further computer readable information about the cause of the
+  alert. The schema of details should be the same for all alerts of the same
+  type. If multiple alerts of a type are possible they should be distiguishable by details.
```

Comment:
What's the intended relation between type/details and information in the status map?